sbom/diff.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# This script runs sbom-generator diff against the committed SBOM.
+# The diff subcommand generates a fresh SBOM internally and compares
+# it against the existing SBOM at --output-path. Exits non-zero on
+# meaningful differences.
+
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+
+WRAPPER="${REPO_ROOT}/scripts/run-from-container.sh"
+IMAGE="sbom-generator-test"
+CONFIG="${REPO_ROOT}/sbom/sbom.config.yaml"
+COMMITTED_SBOM="${REPO_ROOT}/sbom/sbom.cdx.json"
+
+if [[ ! -x "${WRAPPER}" ]]; then
+  echo "Missing or non-executable ${WRAPPER}." >&2
+  exit 1
+fi
+
+# Ensure cache and output directories exist
+XDG_CACHE_HOME="${REPO_ROOT}/.cache"
+mkdir -p "${XDG_CACHE_HOME}"
+mkdir -p "$(dirname "${COMMITTED_SBOM}")"
+
+# Some container images don't have /tmp; set TMPDIR explicitly
+export TMPDIR="${XDG_CACHE_HOME}"
+
+echo "[sbom] Running diff against ${COMMITTED_SBOM} ..."
+extra_env=()
+# Three different potential token-variables are mentioned, we check for all three.
+if [[ -n "${GITHUB_TOKEN:-}" ]]; then
+  extra_env+=(--env GITHUB_TOKEN)
+elif [[ -n "${GH_TOKEN:-}" ]]; then
+  extra_env+=(--env GITHUB_TOKEN="${GH_TOKEN}")
+elif [[ -n "${CK8S_GITHUB_TOKEN:-}" ]]; then
+  extra_env+=(--env GITHUB_TOKEN="${CK8S_GITHUB_TOKEN}")
+fi
+set +e
+"${WRAPPER}" --env XDG_CACHE_HOME="${XDG_CACHE_HOME}" --env TMPDIR="${TMPDIR}" "${extra_env[@]}" \
```

Comment:
```suggestion
"${WRAPPER}" --env XDG_CACHE_HOME="${XDG_CACHE_HOME}" \
  --mount "type=bind,src=${XDG_CACHE_HOME},dst=${XDG_CACHE_HOME}" \
  "${extra_env[@]}" \
```

We would need to mount these paths also for above to work, in all wrappers